docs/gc_overview.md

```diff
@@ -23,81 +23,183 @@
 -->
 
 
-# Memory management overview
+# Garbage collection
 
-The process of managing memory in the VM is handled by the Allocator and the Garbage Collector. These components operate on an area of memory that is reserved for VM processing called the Java&trade; heap.
+To prevent applications running out of memory, objects in the Java heap that are no longer required must be reclaimed. This process is known as *garbage collection* (GC). When garbage is collected, the garbage collector must obtain exclusive access to the heap, which causes an application to pause while the clean up is done. This pause is often referred to as a *stop-the-world* (STW) pause because an application must halt until the process completes. In general, the first step in the GC process is to *mark* the objects that are reachable, which means they are still in use. The next step is to *sweep* away the unmarked objects to reclaim memory. The final step is to *compact* the heap if the heap is badly fragmented.
 
+A GC cycle is a repeatable process that involves a set of GC operations. These operations process all or parts of the Java heap. When operating on the
+whole of the Java heap, the cycle is referred to as a *global GC cycle*; When operating on part of the heap, the cycle is referred to as a *partial GC cycle*.
 
-## The Allocator
+A global GC cycle can be triggered explicitly or implicitly according to the following rules:
 
-The Allocator assigns areas of the Java heap for Java objects. Objects are considered as *live* when they have a chain of references to them that start from root references, such as those found in thread stacks. When that reference or pointer no longer exists, the objects are considered as *garbage*.
+- A global GC cycle is triggered implicitly if it occurs because of internal mechanisms, such as an allocation failure or a taxation threshold being reached.
```

Comment:
You may some examples now, or in elsewhere (in corresponding GC operations). This is a nice comparison:
- taxation threshold being reached is the one use to trigger PGC (since there is not predetermined eden space, we would end up consuming ALL empty regions before hitting AF - we have to trigger PGC earlier using the threshold),
-  allocation failure is used for scavenge (since allocate is predetermined area of heap where allocation threads allocate from, this mechanism works).
 
Some other examples:
The former is also used to trigger GMP increments (between interleaving PGCs)
The latter is used to trigger global GC in optthruput.

Reply:
- Added taxation threshold as the trigger for the balanced PGC in "copy-forward operations" section
- Added allocation failure as the trigger in the "scavenge operation" section
- Added taxation threshold as the trigger to the GMP section of balanced in policy topic
- Added allocation failure as the trigger for optthruput in policy topic.